app/src/test/java/com/github/raulccabreu/redukt/BaseAnnotatedMiddlewareTest.kt

```diff
@@ -193,6 +179,246 @@ class BaseAnnotatedMiddlewareTest {
         }
     }
 
+    @Test
+    fun whenUseBeforeActions() {
+        var result: Pair<String, String>? = null
+        val signal = CountDownLatch(1)
+
+        val beforeMiddleware = ValidMiddlewareActions { state, actionName ->
+            result = Pair(state, actionName)
+            signal.countDown()
+        }
+
+        val redukt = Redukt("initial")
+        redukt.middlewares.add(beforeMiddleware)
+
+        redukt.dispatch(Action("valid", "new state"), false)
+
+        signal.await()
+
+        Assert.assertEquals("initial", result?.first)
+        Assert.assertEquals("valid", result?.second)
+
+        redukt.stop()
+    }
+
+    @Test
+    fun whenUseAfterActions() {
+        var result: Pair<String, String>? = null
+        val signal = CountDownLatch(1)
+
+        val beforeMiddleware = ValidMiddlewareActions ( afterCallback = { state, actionName ->
```

Comment:
Maybe using my suggested middleware in the case of single callback instance too will not affect the test (see suggestion at another review comment)